server/src/main/java/io/crate/role/TransportCreateRoleAction.java

```diff
@@ -120,7 +121,10 @@ static boolean putRole(Metadata.Builder mdBuilder,
         UsersPrivilegesMetadata oldUserPrivilegesMetadata = (UsersPrivilegesMetadata) mdBuilder.getCustom(UsersPrivilegesMetadata.TYPE);
         RolesMetadata newMetadata = RolesMetadata.of(mdBuilder, oldUsersMetadata, oldUserPrivilegesMetadata, oldRolesMetadata);
         boolean exists = true;
-        if (newMetadata.contains(roleName) == false && newMetadata.contains(jwtProperties) == false) {
+        if (newMetadata.contains(jwtProperties)) {
+            throw new RoleAlreadyExistsException("Another role with the same combination of jwt properties already exists");
```

Comment:
If this now raises the `exists` flag above is no longer needed. The function can just return `false`, or change the type to avoid.